tests/unit/test_table.py

```diff
@@ -3550,19 +3550,10 @@ def test__eq___type_mismatch(self):
         self.assertNotEqual(object_under_test, object())
         self.assertEqual(object_under_test, mock.ANY)
 
-    def test___hash__set_equality(self):
+    def test_unhashable_object(self):
         object_under_test1 = self._make_one(start=1, end=10, interval=2)
-        object_under_test2 = self._make_one(start=1, end=10, interval=3)
-        set_one = {object_under_test1, object_under_test2}
-        set_two = {object_under_test1, object_under_test2}
-        self.assertEqual(set_one, set_two)
 
-    def test___hash__not_equals(self):
-        object_under_test1 = self._make_one(start=1, end=10, interval=2)
-        object_under_test2 = self._make_one(start=1, end=10, interval=3)
-        set_one = {object_under_test1}
-        set_two = {object_under_test2}
-        self.assertNotEqual(set_one, set_two)
+        self.assertIsNone(object_under_test1.__hash__)
```

Comment:
I would make this assertion more explicit, i.e. checking that hashing an object produces an error. Something like the following:
```py
with six.assertRaisesRegex(self, TypeError, r".*unhashable type.*"):
    hash(object_under_test1)
```